dialog.cpp

```diff
@@ -109,6 +109,8 @@ Dialog::~Dialog()
     settings.setValue("Filters/ShowInstalled", ui->FltrsWidget->installedChecked());
     settings.setValue("Filters/ShowUpgraded", ui->FltrsWidget->upgradedChecked());
     settings.setValue("Filters/ShowRemoved", ui->FltrsWidget->removedChecked());
+    settings.setValue("Filters/ShowDowngraded", ui->FltrsWidget->downgradedChecked());
+    settings.setValue("Filters/ShowReinstalled", ui->FltrsWidget->reinstalledChecked());
```

Comment:
The added settings should be used in `Dialog::loadSettings()` so that the checked states are preserved after restarting the application.